internal/services/pam/pam_test.go

```diff
@@ -489,12 +489,14 @@ func TestSetDefaultBrokerForUser(t *testing.T) {
 		username string
 
 		// These are auxiliary inputs that affect the test setup and help control the mock output.
-		noBroker bool
+		noBroker    bool
+		localBroker bool
 
 		// This is the expected return.
 		wantErr bool
 	}{
-		"Set default broker for existing user": {username: "usersetbroker"},
+		"Set default broker for existing user":                  {username: "usersetbroker"},
+		"Setting local broker as default should not save on DB": {username: "userlocalbroker", localBroker: true},
```

Comment:
Looking at this, there is one missing case to me: updating user with a new broker (you can use userwithinactivebroker here and set it to the mock broker, wdyt?)

Reply:
I'll add the test case, but we can't use this user (as it would affect other tests, since we use one broker for all of them and they are all in parallel).

Reply:
hum, the broker is indeed global to the package tests. I wonder if that's not going to break us in all those mutable cases in the future. Let’s go with this for now, but it’s something to consider for the future.